File: app/src/main/java/com/app/gamer_shop/models/Product.kt
package com.app.gamer_shop.models

import kotlinx.serialization.Serializable

@Serializable
data class Product(val id: Int, val name: String, val price: Double, val specs: Map<String, Any>,
                   val images: List<String>? = emptyList(),
                   val category: String, val brand: String, val rating: Double)

File: backend/model/product.go
package model

type Product struct {
	Id             int            `json:"id"`
	Name           string         `json:"name"`
	Price          float64        `json:"price"`
	Specifications map[string]any `json:"specs"`
	Images         []string       `json:"images"`
	CategoryName   string         `json:"category"`
	BrandName      string         `json:"brand"`
	AvgRating      float64        `json:"rating"`
}
